app/services/integration/kubernetes_service.rb

```diff
@@ -151,13 +163,50 @@ def initialize(url, service, port)
     end
   end
 
+  class IngressSpec < K8s::Resource
+    def initialize(url, service, port)
+      uri = URI(url)
+      tls_options = [{
+        hosts: [uri.host || uri.path],
+        secretName: service + '-tls'
```

Comment:
I noticed one issue with the tls secretName. There may be collisions for a multi tenant scenario so I will add the tenant name into the secretName.